crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -693,4 +767,142 @@ mod tests {
         let result = super::not_required(Value::from_serialize(&attrs)).unwrap();
         assert_eq!(result.len(), 1);
     }
+
+    #[test]
+    fn test_instantiated_type() {
+        let mut env = Environment::new();
+        let ctx = serde_json::Value::Null;
+
+        add_tests_and_filters(&mut env);
+
+        assert_eq!(
+            env.render_str("{{ 'int' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "int"
+        );
+        assert_eq!(
+            env.render_str("{{ 'int[]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "int[]"
+        );
+        assert_eq!(
+            env.render_str("{{ 'template[double]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "double"
+        );
+        assert_eq!(
+            env.render_str("{{ 'template[double[]]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "double[]"
+        );
+        assert_eq!(
+            env.render_str("{{ 'boolean' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "boolean"
+        );
+        assert_eq!(
+            env.render_str("{{ 'boolean[]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "boolean[]"
+        );
+        assert_eq!(
+            env.render_str("{{ 'template[boolean]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "boolean"
+        );
+        assert_eq!(
+            env.render_str("{{ 'string' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "string"
+        );
+        assert_eq!(
+            env.render_str("{{ 'string[]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "string[]"
+        );
+        assert_eq!(
+            env.render_str("{{ 'template[string]' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "string"
+        );
+        assert_eq!(
+            env.render_str("{{ 'something else' | instantiated_type }}", &ctx)
+                .unwrap(),
+            "enum"
+        );
+        assert_eq!(
+            env.render_str(
+                "{{ {allow_custom_values: true, members: []} | instantiated_type }}",
+                &ctx
+            )
+            .unwrap(),
+            "enum"
```

Comment:
We actually should do better here.  We want to know the *type of the enum*, i.e. are the values of the enum booleans, doubles, ints or strings.

Reply:
I can implement this, although it seems that the Python version didn't do it. Isn't there also a potential stability issue with the APIs? For example, an enum containing only integers will switch to a string representation if we add a string variant to it. This is not necessarily a big problem at the instrumentation level, but it can have implications downstream in the system.

Reply:
Resolved, see [548edf7](https://github.com/open-telemetry/weaver/pull/163/commits/548edf7d62912674e37b16d0c0672def45c83d8c)